db/controllers/comments.controller.js

```diff
@@ -7,9 +7,29 @@ exports.getCommentsByArticleId = (req, res, next) => {
         if (comments.length === 0) {
             res.status(204).send();
         }
-      res.status(200).send({ comments });
+        res.status(200).send({ comments });
     })
     .catch((err) => {
       next(err);
     })
-  };
\ No newline at end of file
+};
+
+exports.postCommentByArticleId = (req, res, next) => {
+    const { article_id } = req.params;
+    const { username, body } = req.body;
+
+    if (!username || !body) {
+        return res.status(400).send({ msg: "Bad request - must have a valid username and a body" });
+    }
+
+    addComment(article_id, username, body)
+        .then((comment) => {
+            if (!comment) {
+                return Promise.reject({ status: 400 });
+            }
+            res.status(201).send({ comment });
+        })
```

Comment:
Same as the above comment, but also we should always assume by entering this .then(...) we are following the happy path